samples/cxx/custom/custom.cpp

```diff
@@ -0,0 +1,197 @@
+
+    /* ---------------------- CREATE CSV OUTPUT FILE ---------------------- */
+    // simulation results will be outputted to a .csv file as complete state vectors
+    // for each simulation time point.
+    // create the csv file, overwriting any existing ones with the same name.
+    std::ofstream outputFile("custom_cxx.csv");
+    
+    // for convenience, a title for each of the state vector's components is written to
+    // the first line of the csv file.
+    outputFile << "time (s), temp (K), ";
+    for (int k = 0; k < gas->nSpecies(); k++)
+        outputFile << "Y_" << gas->speciesName(k) << ", ";
+    outputFile << std::endl;
+
+    /* --------------------- CREATE ODE RHS EVALUATOR --------------------- */
+    ReactorODEs odes = ReactorODEs(sol);
+
+    /* ---------------------- SPECIFY TIME INTERVAL ----------------------- */
+    // the simulation is run over the time interval specified below. tnow is initialized
+    // with the simulation's start time, and is updated on each timestep to reflect
+    // the new absolute time of the system.
+    double tnow = 0.0;
+    double tfinal = 1e-3;
+    
+    /* ------------------- CREATE & INIT ODE INTEGRATOR ------------------- */
+    // create an ODE integrator object, which will be used to solve the system of ODES defined
+    // in the ReactorODEs class. a C++ interface to the C-implemented SUNDIALS CVODES integrator
+    // (CVodesIntegrator) is built into Cantera, and will be used to solve this example.
+    //  - the default settings for CVodesIntegrator are used:
+    //     solution method: BDF_Method
+    //     problem type: DENSE + NOJAC
+    //     relative tolerance: 1.0e-9
+    //     absolute tolerance: 1.0e-15
+    //     max step size: +inf
+    Integrator *integrator = newIntegrator("CVODE");
+
+    // initialize the integrator, specifying the start time and the RHS evaluator object.
+    // internally, the integrator will apply settings, allocate needed memory, and populate
+    // this memory with the appropriate initial values for the system.
+    integrator->initialize(tnow, odes);
+    
+    /* ----------------------- SIMULATION TIME LOOP ----------------------- */
+    while (tnow < tfinal) {
+        // advance the simulation to the current absolute time, tnow, using the integrator's
+        // ODE system time-integration capability. a pointer to the resulting system state vector
+        // is fetched in order to access the solution components.
+        integrator->integrate(tnow);
+        double *solution = integrator->solution();
+
+        // output the current absolute time and solution state vector to the csv file. you can view
+        // these results by opening the "custom_cxx.csv" file that appears in this program's directory
+        // after compiling and running.
+        outputFile << tnow << ", ";
+        for (int i = 0; i < odes.neq(); i++)
+            outputFile << solution[i] << ", ";
+        outputFile << std::endl;
```

Comment:
In this case, there's a fairly easy way to avoid the trailing comma after the last variable on each line:
```suggestion
        outputFile << tnow;
        for (int i = 0; i < odes.neq(); i++) {
            outputFile << ", " << solution[i];
        }
        outputFile << std::endl;
```